Lib/test/test_py3kwarn.py

```diff
@@ -112,6 +112,60 @@ def test_dict_inequality_comparisons(self):
             w.reset()
             self.assertWarning({2:3} >= {}, w, expected)
 
+    def test_dict_keys(self):
+        expected = 'dict.keys() returns a view in 3.x: convert the result to a list'
```

Comment:
Doesn't it depend on where `keys` is used? For example in:

```python
for x in <dict>.keys():
```

There's no point adding `list(keys)` but if you do `x = <dict>.keys()` you probably did want `x = list(<dict>.keys())`?

Reply:
Yes it does, do we need to reword the warning?

Reply:
We can never have a perfect warning, but I think we can perhaps use the syntactic context (or some proxy thereof) to give a pretty good approximation. For example if we add a warning for every `for x in <dict>.keys():` it's not only going to overwhelm users but telling them to add `list(...)` will make their code worse in many situations. Perhaps we could say, for example, something like "if we spot the `keys` method and the next opcode is `Assign` then issue a warning"? [I'm sure that's not quite right, and it might also be difficult to implement, but it might give us an idea of what we'd like to see.]

Reply:
I agree, is function the scope again? or we scanning the whole file for these conditions?

Reply:
Maybe we can just say "if you call `keys` in the condition of a `for`/`while` loop then don't warn, otherwise warn"?